gh_actions_exporter/cost.py

```diff
@@ -0,0 +1,123 @@
+import jinja2
+
+from typing import Dict, List
+from gh_actions_exporter.config import Relabel, RelabelType, Settings
+from gh_actions_exporter.types import WebHook, JobCost
+from gh_actions_exporter.tokenGenerator import TokenGenerator
+from github import Github, Repository
+
+class Cost(object):
+    def __init__(self, settings: Settings):
+        self.settings: Settings = settings 
+
+    def relabel_job_names_dict(self, relabel: Relabel, job_request: dict) -> dict:
+        result: dict = {
+            relabel.label: relabel.default
+        }
+        if job_request["status"] == 'queued':
+            result[relabel.label] = ""
+        else:
+            for label in relabel.values:
+                if label in job_request["runner_name"]:
+                    result[relabel.label] = label
+        return result
+
+    def runner_type_job(self, job_request: dict) -> str:
+        if 'self-hosted' in job_request["labels"]:
+            return 'self-hosted'
+        return 'github-hosted'
+    
+    def relabel_job_labels(self, relabel: Relabel, labels: List[str]) -> Dict[str, str or None]:
+        result: dict = {
+            relabel.label: relabel.default
+        }
+        for label in relabel.values:
+            if label in labels:
+                result[relabel.label] = label
+        return result
+
+    def job_request_labels(self, job_request: dict, webhook: WebHook, settings: Settings) -> dict:
+        labels: dict = dict(
+            runner_type=self.runner_type_job(job_request),
+            job_name=job_request["name"],
+            repository_visibility=webhook.workflow_run.repository.private,
+            repository=webhook.workflow_run.repository.full_name,
+        )
+        for relabel in settings.job_relabelling:
+            if relabel.type == RelabelType.label:
+                labels.update(self.relabel_job_labels(relabel, job_request["labels"]))
+            elif relabel.type == RelabelType.name:
+                labels.update(self.relabel_job_names_dict(relabel, job_request))
+        return labels
+
+    def _get_job_cost(self, job_request: dict, webhook: WebHook) -> float:
+        labels: dict = self.job_request_labels(job_request, webhook, self.settings)
+        flavor = labels.get(self.settings.flavor_label, None)
+        cost_per_min: float = self.settings.job_costs.get(flavor, self.settings.default_cost)
+        duration: float = (job_request["completed_at"].timestamp()
```

Comment:
job request is not typed. It seems like a [WorkflowJob](https://github.com/scality/gh-actions-exporter/pull/20/files/4dde2b798b1473e1b66bf462f1faf1d144931549..3e7762f1f426b80668bb0f617366c74606c98c69#diff-492b94ab9c36f44dc4552990feb27270cf5cf3f035de199e212362e0260cc9c9R16) type of object. If so, use it.